src/Symfony/Component/Uid/Ulid.php

```diff
@@ -136,9 +144,13 @@ private static function generate(): string
             self::$rand = array_values($r);
             self::$time = $time;
         } elseif ([0xFFFFF, 0xFFFFF, 0xFFFFF, 0xFFFFF] === self::$rand) {
-            usleep(100);
+            if (null === $timestamp) {
+                usleep(100);
+            } else {
+                self::$time = '';
```

Comment:
Not incrementing the time is going to be problematic, as it means that we will possibly generate duplicate values, especially here, where we exhausted the random space.
I'm not sure it would be an issue to keep the usleep in place.
Alternatively, we could force self::$rand to overflow to `[0, 0, 0, 0]`. This might have my preference here.